File: 69-sqrt_x/solution.py
class Solution:
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        # 100%
        '''
        return int(x**0.5)
        '''
        
        # 45%   binary search
        l, h = 1, x
        
        while l<=h:
            m = l+ (h-l)//2
            sqrt=x/m
            if sqrt==m:
                # print(1)
                return m            
            elif sqrt>m:
                # print(2)
                l=m+1
            else:
                # print(3)
                h=m-1
                
        return int(h)

File: 69-sqrt_x/solution.java
class Solution {
    public int mySqrt(int x) {
        
        // 90%
        // return (int)Math.sqrt(x);
        
        // 90%
        int l=1, h=x;
        int m, sqrt;
        while(l<=h){
            m = l + (h-l)/2;
            sqrt = x/m;
            // System.out.println("sqrt:"+sqrt);
            if(sqrt==m){
                // System.out.println(1);
                return m;
            }else if(sqrt>m){
                // System.out.println(2);
                l=m+1;
            }else{
                // System.out.println(3);
                h=m-1;
            }
        }
        return h;
    }
}